script.js

```diff
@@ -0,0 +1,265 @@
+let array = [
+  {
+    title: "Inception",
+    year: 2010,
+    rating: 8.8,
+    genre: ["Action", "Sci-Fi"],
+    director: "Christopher Nolan",
+  },
+  {
+    title: "The Dark Knight",
+    year: 2008,
+    rating: 9.0,
+    genre: ["Action", "Crime", "Drama"],
+    director: "Christopher Nolan",
+  },
+  {
+    title: "Pulp Fiction",
+    year: 1994,
+    rating: 8.9,
+    genre: ["Crime", "Drama"],
+    director: "Quentin Tarantino",
+  },
+  {
+    title: "The Matrix",
+    year: 1999,
+    rating: 8.7,
+    genre: ["Action", "Sci-Fi"],
+    director: "Lana Wachowski, Lilly Wachowski",
+  },
+  {
+    title: "Fight Club",
+    year: 1999,
+    rating: 8.8,
+    genre: ["Drama"],
+    director: "David Fincher",
+  },
+];
+
+// Create an array of movie titles.
+const getMovieTitle = (array) => {
+  let movieTitles = [];
+
+  for (let value of array) {
+    movieTitles.push(value.title);
+  }
+
+  return movieTitles;
+};
+
+getMovieTitle(array);
+
+//Simple approach
+const getMovieTitles = (array) => array.map((movie) => movie.title);
+getMovieTitles(array);
+
+
+
+// Create an array of movies with a rating higher than 8.5.
+const isHigherRating = (value) => {
+  return value >= 8.5;
+};
+
+const moviesRating = (array) => {
+  let result = [];
+
+  for (let value of array) {
+    array.filter(isHigherRating);
+    result.push(value.rating);
+  }
+  return result;
+};
+
+moviesRating(array);
+
+//Simple approach
+const moviesRatingSimple = (array) => {
+  return array.filter((movie) => movie.rating > 8.5);
+};
+
+moviesRatingSimple(array);
+
+
+
+// Sort the movies by rating in descending order.
+
+const moviesDescending = (array) => {
+  return array.map((movie) => movie.rating).sort((a, b) => (a > b ? -1 : 1));
+};
+
+moviesDescending(array);
+
+
+
+// Create an array of movies released after the year 2000.
+const getMoviesReleasedAfter2000 = (array) => {
+  return array.filter((movie) => movie.year > 2000).map((movie) => movie.title);
+};
+
+getMoviesReleasedAfter2000(array);
+
+
+
+// Sort the movies by year of release in ascending order.
+const moviesAscending = (array) => {
+  return array.map((movie) => movie.year).sort((a, b) => (a > b ? 1 : -1));
+};
+
+moviesAscending(array);
+
+
+
+// Create an object that counts the number of movies for each genre.
+const countMoviesByGenre = (array) => {
+  return array.reduce((acc, movie) => {
+    movie.genre.forEach(genre => {
+      acc[genre] = (acc[genre] || 0) + 1;
+    });
+    return acc;
+  }, {});
+};
+
+countMoviesByGenre(array);
+
+
+
+// Check if all movies have a rating higher than 7.
+const moviesRatingHigher7 = (array) => {
+  return array.every((movie) => movie.rating > 7);
+};
+
+moviesRatingHigher7(array);
+
+
+
+// Calculate the average rating of all movies.
+const averageRating = (array) => {
+  return array.reduce((acc, movie, __, { length }) => acc + movie.rating / length, 0);
```

Comment:
i like what you did here with the length :D

Reply:
Thank you, your feedback is very important and motivating for me)